mfs/inode.go

```diff
@@ -0,0 +1,46 @@
+package mfs
+
+import (
+	ipld "gx/ipfs/QmWi2BYBL5gJ3CiAiQchg6rn1A8iBsrWy51EYxvHVjFvLb/go-ipld-format"
+)
+
+// Inode abstracts the common characteristics of MFS
+// `File` and `Directory`. All of its attributes are
+// initialized at creation and can't be modified later.
+type Inode struct {
+
+	// Name of this `Inode` in the MFS path (the same value is
+	// also stored as the name of the DAG link).
+	name string
+
+	// Parent directory of this `Inode` (which may be the `Root`).
+	parent childCloser
+
+	// DAG service used to store modifications made to the contents
+	// of the file or directory the `Inode` belongs to.
+	dagService ipld.DAGService
+}
+
+// NewInode creates a new `Inode` structure and returns its pointer.
+func NewInode(name string, parent childCloser, dagService ipld.DAGService) Inode {
+	return Inode{
+		name:       name,
+		parent:     parent,
+		dagService: dagService,
+	}
+}
+
+// GetName returns the `Inode`'s name.
+func (i *Inode) GetName() string {
```

Comment:
Why not access these directly? Would be cleaner imo (and look less than java)

Reply:
> Why not access these directly?

To make it explicit that it is a read-only attribute and should not be modified.

> (and look less than java)

Would you prefer just using `Name()`?

https://golang.org/doc/effective_go.html#Getters